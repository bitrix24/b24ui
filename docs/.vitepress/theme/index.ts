/**
 * This is custom theme
 * We change some css
 * @see node_modules/vitepress/dist/client/theme-default/index.js
 */
import { ref, watch, toValue } from 'vue'
import { createRouter, createMemoryHistory } from 'vue-router'
import bitrix24UIPlugin from '@bitrix24/b24ui-nuxt/vue-plugin'
import Layout from './components/ui/Layout.vue'
import ComponentShowExample from './components/ui/ComponentShowExample.vue'
import ComponentProps from './components/ui/ComponentProps.vue'
import ComponentSlots from './components/ui/ComponentSlots.vue'
import ComponentEmits from './components/ui/ComponentEmits.vue'
import Description from './components/ui/Description.vue'
import Placeholder from './components/ui/Placeholder.vue'
import VPBadge from 'vitepress/dist/client/theme-default/components/VPBadge.vue'
import 'vitepress/dist/client/theme-default/styles/fonts.css'
import './tailwind.css'
// region ory theme ////
import './styles/components/custom-block.css'
import './styles/components/vp-code.css'
import './styles/components/vp-code-group.css'
import './styles/components/vp-doc.css'
import './styles/components/vp-sponsor.css'
// endregion ////
import './theme.css'

globalThis.useFetch = async (url: string, options: RequestInit & { params?: any, transform?: (data) => any } = {}) => {
  const data = ref()
  const status = ref('idle')
  async function _fetch() {
    status.value = 'loading'
    try {
      const params = options.params
        ? Object.fromEntries(
            Object.entries(options.params).map(([key, value]) => [key, toValue(value)])
          )
        : {}

      const queryString = new URLSearchParams(params as any).toString()
      const fullUrl = queryString ? `${url}?${queryString}` : url

      data.value = await fetch(fullUrl, options).then(r => r.json()).then(r => options.transform ? options.transform(r) : r)
      status.value = 'success'
    } catch (error) {
      console.error(error)
      status.value = 'error'
    }
  }
  _fetch()

  if (options?.params?.q) {
    watch(
      options.params.q,
      () => {
        _fetch()
      },
      { deep: true }
    )
  }

  return Promise.resolve({
    data,
    status
  })
}

export default {
  Layout,
  enhanceApp: async ({ app }) => {
    // region components ////
    app.component('Badge', VPBadge)
      .component('Description', Description)
      .component('ComponentShowExample', ComponentShowExample)
      .component('ComponentProps', ComponentProps)
      .component('ComponentSlots', ComponentSlots)
      .component('ComponentEmits', ComponentEmits)
      .component('Placeholder', Placeholder)
    // endregion ////

    if (!import.meta.env.SSR) {
      const router = createRouter({
        routes: [
          // { path: '', component: app }
        ],
        history: createMemoryHistory()
        // history: createWebHistory()
      })
      app.use(router)

      app.use(bitrix24UIPlugin)
    }
  }
}
