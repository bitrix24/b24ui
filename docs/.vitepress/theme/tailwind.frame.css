/*
@plugin '@bitrix24/b24style';
@plugin "@bitrix24/b24ui-nuxt";
*/
@import "tailwindcss" source("../../");
@import "@bitrix24/b24ui-nuxt";
/*
*/



