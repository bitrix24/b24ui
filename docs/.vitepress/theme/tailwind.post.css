@import "tailwindcss" source("../../../docs");
@import "@bitrix24/b24ui-nuxt";

/**
 * bxshef: Custom
 * -------------------------------------------------------------------------- */

:root
{
}

.dark
{
  --vp-c-bg: #0f172a;
  --vp-c-bg-alt: #161618;
  --vp-c-bg-elv: #202127;
  --vp-c-bg-soft: #202127;
}

.bg-grid-example {
  background-image: url("data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 32 32' width='32' height='32' fill='none' stroke='rgb(130 139 149 / 0.25)'%3e%3cpath d='M0 .5H31.5V32'/%3e%3c/svg%3e");
}

.dark\:bg-grid-example:is(.dark *) {
  background-image: url("data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 32 32' width='32' height='32' fill='none' stroke='rgb(130 139 149 / 0.25)'%3e%3cpath d='M0 .5H31.5V32'/%3e%3c/svg%3e");
}
