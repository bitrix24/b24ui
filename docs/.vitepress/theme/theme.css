/**
 * bxshef: Custom
 * -------------------------------------------------------------------------- */

:root
{
  --sh-scrollbar-thumb: #3c3c43;
  --sh-scrollbar-background: #ebebef;
}

.dark
{
  --vp-c-bg: #0f172a;
  --vp-c-bg-alt: #161618;
  --vp-c-bg-elv: #202127;
  --vp-c-bg-soft: #202127;
  --vp-code-block-bg: #1a2234;

  --sh-scrollbar-thumb: #bdc1c6;
  --sh-scrollbar-background: #3c3c43;
}

.bg-grid-example {
  background-image: url("data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 32 32' width='32' height='32' fill='none' stroke='rgb(130 139 149 / 0.25)'%3e%3cpath d='M0 .5H31.5V32'/%3e%3c/svg%3e");
}

.dark\:bg-grid-example:is(.dark *) {
  background-image: url("data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 32 32' width='32' height='32' fill='none' stroke='rgb(130 139 149 / 0.25)'%3e%3cpath d='M0 .5H31.5V32'/%3e%3c/svg%3e");
}

iframe
{
  min-width: 100%;
  border: none;
  scrollbar-width: thin;
  background-color: transparent;
}

/** region scrollbar */
* {
  scrollbar-color: var(--sh-scrollbar-thumb) var(--sh-scrollbar-background);
  scrollbar-width: thin;
}

/* Styling the scrollbar in Chrome and Safari */
*::-webkit-scrollbar {
  width: 6px;
}

/* Set thumb color for Chrome and Safari */
*::-webkit-scrollbar-thumb {
  background-color: var(--sh-scrollbar-thumb);
  border-radius: 4px;
}

/* Set track color for Chrome and Safari */
*::-webkit-scrollbar-track {
  background-color: var(--sh-scrollbar-background);
}
/** endregion */
