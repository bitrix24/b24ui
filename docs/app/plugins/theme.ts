export default defineNuxtPlugin({
  enforce: 'post',
  setup() {
  }
})
