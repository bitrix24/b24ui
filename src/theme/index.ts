export { default as advice } from './advice'
export { default as avatar } from './avatar'
export { default as avatarGroup } from './avatar-group'
export { default as badge } from './badge'
export { default as button } from './button'
export { default as buttonGroup } from './button-group'
export { default as checkbox } from './checkbox'
export { default as chip } from './chip'
export { default as container } from './container'
export { default as countdown } from './countdown'
export { default as formField } from './form-field'
export { default as input } from './input'
export { default as kbd } from './kbd'
export { default as link } from './link'
export { default as progress } from './progress'
export { default as radioGroup } from './radio-group'
export { default as range } from './range'
export { default as separator } from './separator'
export { default as skeleton } from './skeleton'
export { default as switch } from './switch'
export { default as textarea } from './textarea'
export { default as tooltip } from './tooltip'
