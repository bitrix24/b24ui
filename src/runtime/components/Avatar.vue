<script lang="ts">
import type { DefineComponent } from 'vue'
import type { VariantProps } from 'tailwind-variants'
import type { AvatarFallbackProps } from 'reka-ui'
import type { AppConfig } from '@nuxt/schema'
import _appConfig from '#build/app.config'
import theme from '#build/b24ui/avatar'
import { tv } from '../utils/tv'

const appConfig = _appConfig as AppConfig & { b24ui: { avatar: Partial<typeof theme> } }

const avatar = tv({ extend: tv(theme), ...(appConfig.b24ui?.avatar || {}) })

type AvatarVariants = VariantProps<typeof avatar>

export interface AvatarProps extends Pick<AvatarFallbackProps, 'delayMs'> {
  /**
   * The element or component this component should render as.
   * @defaultValue 'span'
   */
  as?: any
  src?: string
  alt?: string
  icon?: DefineComponent
  text?: string
  size?: AvatarVariants['size']
  class?: any
  b24ui?: Partial<typeof avatar.slots>
}

export interface AvatarSlots {
  default(props?: {}): any
}
</script>

<script setup lang="ts">
import { computed } from 'vue'
import { AvatarRoot, AvatarImage, AvatarFallback, useForwardProps } from 'reka-ui'
import { reactivePick } from '@vueuse/core'
import { useAvatarGroup } from '../composables/useAvatarGroup'
import ImageComponent from '#build/b24ui-image-component'

defineOptions({ inheritAttrs: false })

const props = withDefaults(defineProps<AvatarProps>(), { as: 'span' })

const fallbackProps = useForwardProps(reactivePick(props, 'delayMs'))

const fallback = computed(() => props.text || (props.alt || '')
  .replace(/[\+\-\*\)\(\}\]\[\{\}]/g, '')
  .split(' ')
  .map(word => word.charAt(0))
  .join('')
  .substring(0, 2)
)

const { size } = useAvatarGroup(props)

// eslint-disable-next-line vue/no-dupe-keys
const b24ui = computed(() => avatar({
  size: size.value
}))

const sizePx = computed(() => ({
  '2xs': 14,
  'xs': 18,
  'sm': 22,
  'md': 32,
  'lg': 42,
  'xl': 48,
  '2xl': 60,
  '3xl': 94
})[props.size || 'md'])
</script>

<template>
  <AvatarRoot :as="as" :class="b24ui.root({ class: [props.class, props.b24ui?.root] })">
    <AvatarImage
      v-if="src"
      :as="ImageComponent"
      :src="src"
      :alt="alt"
      :width="sizePx"
      :height="sizePx"
      v-bind="$attrs"
      :class="b24ui.image({ class: props.b24ui?.image })"
    />

    <AvatarFallback as-child v-bind="{ ...fallbackProps, ...$attrs }">
      <slot>
        <Component :is="icon" v-if="icon" :class="b24ui.icon({ class: props.b24ui?.icon })" />
        <span v-else :class="b24ui.fallback({ class: props.b24ui?.fallback })">{{ fallback || '&nbsp;' }}</span>
      </slot>
    </AvatarFallback>
  </AvatarRoot>
</template>
