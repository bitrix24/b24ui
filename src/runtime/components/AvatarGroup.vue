<script lang="ts">
import type { AppConfig } from '@nuxt/schema'
import theme from '#build/b24ui/avatar-group'
import type { ComponentConfig } from '../types/tv'

type AvatarGroup = ComponentConfig<typeof theme, AppConfig, 'avatarGroup'>

export interface AvatarGroupProps {
  /**
   * The element or component this component should render as.
   * @defaultValue 'div'
   */
  as?: any
  /**
   * @defaultValue 'md'
   */
  size?: AvatarGroup['variants']['size']
  /**
   * The maximum number of avatars to display.
   */
  max?: number | string
  class?: any
  b24ui?: AvatarGroup['slots']
}

export interface AvatarGroupSlots {
  default(props?: {}): any
}
</script>

<script setup lang="ts">
import { computed, provide } from 'vue'
import { Primitive } from 'reka-ui'
import { useAppConfig } from '#imports'
import { avatarGroupInjectionKey } from '../composables/useAvatarGroup'
import { tv } from '../utils/tv'
import B24Avatar from './Avatar.vue'

const props = defineProps<AvatarGroupProps>()
const slots = defineSlots<AvatarGroupSlots>()

const appConfig = useAppConfig() as AvatarGroup['AppConfig']

const b24ui = computed(() => tv({ extend: tv(theme), ...(appConfig.b24ui?.avatarGroup || {}) })({
  size: props.size
}))

const max = computed(() => typeof props.max === 'string' ? Number.parseInt(props.max, 10) : props.max)

const children = computed(() => {
  let children = slots.default?.()
  if (children?.length) {
    children = children.flatMap((child: any) => {
      if (typeof child.type === 'symbol') {
        // `v-if="false"` or commented node
        if (typeof child.children === 'string') {
          return
        }

        return child.children
      }

      return child
    }).filter(Boolean)
  }

  return children || []
})

const visibleAvatars = computed(() => {
  if (!children.value.length) {
    return []
  }

  if (!max.value || max.value <= 0) {
    return [...children.value]
  }

  return [...children.value].slice(0, max.value)
})

const hiddenCount = computed(() => {
  if (!children.value.length) {
    return 0
  }

  return children.value.length - visibleAvatars.value.length
})

provide(avatarGroupInjectionKey, computed(() => ({
  size: props.size
})))
</script>

<template>
  <Primitive :as="as" :class="b24ui.root({ class: [props.b24ui?.root, props.class] })">
    <component :is="avatar" v-for="(avatar, count) in visibleAvatars" :key="count" :class="b24ui.base({ class: props.b24ui?.base })" />
    <B24Avatar v-if="hiddenCount > 0" :text="`+${hiddenCount}`" :class="b24ui.base({ class: props.b24ui?.base })" />
  </Primitive>
</template>
