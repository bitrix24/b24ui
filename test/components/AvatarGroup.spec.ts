import { defineComponent } from 'vue'
import { describe, it, expect } from 'vitest'
import Avatar from '../../src/runtime/components/Avatar.vue'
import AvatarGroup, { type AvatarGroupProps, type AvatarGroupSlots } from '../../src/runtime/components/AvatarGroup.vue'
import ComponentRender from '../component-render'
import theme from '#build/b24ui/avatar-group'

const AvatarGroupWrapper = defineComponent({
  components: {
    B24Avatar: Avatar,
    B24AvatarGroup: AvatarGroup
  },
  template: `<B24AvatarGroup>
  <B24Avatar src="https://github.com/IgorShevchik.png" alt="Igor Shevchik" />
  <B24Avatar src="https://github.com/romhml.png" alt="Romain Hamel" />
  <B24Avatar src="https://github.com/noook.png" alt="Neil Richter" />
</B24AvatarGroup>`
})

describe('AvatarGroup', () => {
  const sizes = Object.keys(theme.variants.size) as any

  it.each([
    // Props
    ['with max', { props: { max: 2 } }],
    ...sizes.map((size: string) => [`with size ${size}`, { props: { size } }]),
    ['with as', { props: { as: 'span' } }],
    ['with class', { props: { class: 'justify-start' } }],
    ['with b24ui', { props: { b24ui: { base: 'rounded-lg' } } }],
    // Slots
    ['with default slot', {}]
  ])('renders %s correctly', async (nameOrHtml: string, options: { props?: AvatarGroupProps, slots?: Partial<AvatarGroupSlots> }) => {
    const html = await ComponentRender(nameOrHtml, options, AvatarGroupWrapper)
    expect(html).toMatchSnapshot()
  })
})
