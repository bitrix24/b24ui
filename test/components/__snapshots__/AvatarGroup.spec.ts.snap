// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AvatarGroup > renders with as correctly 1`] = `
"<span class="inline-flex justify-end"><span class="inline-flex items-center justify-center shrink-0 select-none overflow-hidden align-middle bg-base-100 dark:bg-base-900 size-8 text-sm relative rounded-full ring-white dark:ring-base-700 last:me-0 ring-2 -me-1.5"><img role="img" src="https://github.com/IgorShevchik.png" alt="Igor Shevchik" width="32" height="32" class="h-full w-full rounded-[inherit] object-cover"></span><span class="inline-flex items-center justify-center shrink-0 select-none overflow-hidden align-middle bg-base-100 dark:bg-base-900 size-8 text-sm relative rounded-full ring-white dark:ring-base-700 last:me-0 ring-2 -me-1.5"><img role="img" src="https://github.com/romhml.png" alt="Romain Hamel" width="32" height="32" class="h-full w-full rounded-[inherit] object-cover"></span><span class="inline-flex items-center justify-center shrink-0 select-none overflow-hidden align-middle bg-base-100 dark:bg-base-900 size-8 text-sm relative rounded-full ring-white dark:ring-base-700 last:me-0 ring-2 -me-1.5"><img role="img" src="https://github.com/noook.png" alt="Neil Richter" width="32" height="32" class="h-full w-full rounded-[inherit] object-cover"></span>
<!--v-if--></span>"
`;

exports[`AvatarGroup > renders with b24ui correctly 1`] = `
"<div class="inline-flex justify-end"><span class="inline-flex items-center justify-center shrink-0 select-none overflow-hidden align-middle bg-base-100 dark:bg-base-900 size-8 text-sm relative ring-white dark:ring-base-700 last:me-0 ring-2 -me-1.5 rounded-lg"><img role="img" src="https://github.com/IgorShevchik.png" alt="Igor Shevchik" width="32" height="32" class="h-full w-full rounded-[inherit] object-cover"></span><span class="inline-flex items-center justify-center shrink-0 select-none overflow-hidden align-middle bg-base-100 dark:bg-base-900 size-8 text-sm relative ring-white dark:ring-base-700 last:me-0 ring-2 -me-1.5 rounded-lg"><img role="img" src="https://github.com/romhml.png" alt="Romain Hamel" width="32" height="32" class="h-full w-full rounded-[inherit] object-cover"></span><span class="inline-flex items-center justify-center shrink-0 select-none overflow-hidden align-middle bg-base-100 dark:bg-base-900 size-8 text-sm relative ring-white dark:ring-base-700 last:me-0 ring-2 -me-1.5 rounded-lg"><img role="img" src="https://github.com/noook.png" alt="Neil Richter" width="32" height="32" class="h-full w-full rounded-[inherit] object-cover"></span>
  <!--v-if-->
</div>"
`;

exports[`AvatarGroup > renders with class correctly 1`] = `
"<div class="inline-flex justify-start"><span class="inline-flex items-center justify-center shrink-0 select-none overflow-hidden align-middle bg-base-100 dark:bg-base-900 size-8 text-sm relative rounded-full ring-white dark:ring-base-700 last:me-0 ring-2 -me-1.5"><img role="img" src="https://github.com/IgorShevchik.png" alt="Igor Shevchik" width="32" height="32" class="h-full w-full rounded-[inherit] object-cover"></span><span class="inline-flex items-center justify-center shrink-0 select-none overflow-hidden align-middle bg-base-100 dark:bg-base-900 size-8 text-sm relative rounded-full ring-white dark:ring-base-700 last:me-0 ring-2 -me-1.5"><img role="img" src="https://github.com/romhml.png" alt="Romain Hamel" width="32" height="32" class="h-full w-full rounded-[inherit] object-cover"></span><span class="inline-flex items-center justify-center shrink-0 select-none overflow-hidden align-middle bg-base-100 dark:bg-base-900 size-8 text-sm relative rounded-full ring-white dark:ring-base-700 last:me-0 ring-2 -me-1.5"><img role="img" src="https://github.com/noook.png" alt="Neil Richter" width="32" height="32" class="h-full w-full rounded-[inherit] object-cover"></span>
  <!--v-if-->
</div>"
`;

exports[`AvatarGroup > renders with default slot correctly 1`] = `
"<div class="inline-flex justify-end"><span class="inline-flex items-center justify-center shrink-0 select-none overflow-hidden align-middle bg-base-100 dark:bg-base-900 size-8 text-sm relative rounded-full ring-white dark:ring-base-700 last:me-0 ring-2 -me-1.5"><img role="img" src="https://github.com/IgorShevchik.png" alt="Igor Shevchik" width="32" height="32" class="h-full w-full rounded-[inherit] object-cover"></span><span class="inline-flex items-center justify-center shrink-0 select-none overflow-hidden align-middle bg-base-100 dark:bg-base-900 size-8 text-sm relative rounded-full ring-white dark:ring-base-700 last:me-0 ring-2 -me-1.5"><img role="img" src="https://github.com/romhml.png" alt="Romain Hamel" width="32" height="32" class="h-full w-full rounded-[inherit] object-cover"></span><span class="inline-flex items-center justify-center shrink-0 select-none overflow-hidden align-middle bg-base-100 dark:bg-base-900 size-8 text-sm relative rounded-full ring-white dark:ring-base-700 last:me-0 ring-2 -me-1.5"><img role="img" src="https://github.com/noook.png" alt="Neil Richter" width="32" height="32" class="h-full w-full rounded-[inherit] object-cover"></span>
  <!--v-if-->
</div>"
`;

exports[`AvatarGroup > renders with max correctly 1`] = `"<div class="inline-flex justify-end"><span class="inline-flex items-center justify-center shrink-0 select-none overflow-hidden align-middle bg-base-100 dark:bg-base-900 size-8 text-sm relative rounded-full ring-white dark:ring-base-700 last:me-0 ring-2 -me-1.5"><img role="img" src="https://github.com/IgorShevchik.png" alt="Igor Shevchik" width="32" height="32" class="h-full w-full rounded-[inherit] object-cover"></span><span class="inline-flex items-center justify-center shrink-0 select-none overflow-hidden align-middle bg-base-100 dark:bg-base-900 size-8 text-sm relative rounded-full ring-white dark:ring-base-700 last:me-0 ring-2 -me-1.5"><img role="img" src="https://github.com/romhml.png" alt="Romain Hamel" width="32" height="32" class="h-full w-full rounded-[inherit] object-cover"></span><span class="inline-flex items-center justify-center shrink-0 select-none overflow-hidden align-middle bg-base-100 dark:bg-base-900 size-8 text-sm relative rounded-full ring-white dark:ring-base-700 last:me-0 ring-2 -me-1.5"><span class="font-medium text-base-500 dark:text-base-600 truncate">+1</span></span></div>"`;

exports[`AvatarGroup > renders with size 2xl correctly 1`] = `
"<div class="inline-flex justify-end"><span class="inline-flex items-center justify-center shrink-0 select-none overflow-hidden align-middle bg-base-100 dark:bg-base-900 size-[60px] text-5xl relative rounded-full ring-white dark:ring-base-700 last:me-0 ring-3 -me-2"><img role="img" src="https://github.com/IgorShevchik.png" alt="Igor Shevchik" width="32" height="32" class="h-full w-full rounded-[inherit] object-cover"></span><span class="inline-flex items-center justify-center shrink-0 select-none overflow-hidden align-middle bg-base-100 dark:bg-base-900 size-[60px] text-5xl relative rounded-full ring-white dark:ring-base-700 last:me-0 ring-3 -me-2"><img role="img" src="https://github.com/romhml.png" alt="Romain Hamel" width="32" height="32" class="h-full w-full rounded-[inherit] object-cover"></span><span class="inline-flex items-center justify-center shrink-0 select-none overflow-hidden align-middle bg-base-100 dark:bg-base-900 size-[60px] text-5xl relative rounded-full ring-white dark:ring-base-700 last:me-0 ring-3 -me-2"><img role="img" src="https://github.com/noook.png" alt="Neil Richter" width="32" height="32" class="h-full w-full rounded-[inherit] object-cover"></span>
  <!--v-if-->
</div>"
`;

exports[`AvatarGroup > renders with size 2xs correctly 1`] = `
"<div class="inline-flex justify-end"><span class="inline-flex items-center justify-center shrink-0 select-none overflow-hidden align-middle bg-base-100 dark:bg-base-900 size-5 text-4xs font-regular relative rounded-full ring-white dark:ring-base-700 last:me-0 ring -me-0.5"><img role="img" src="https://github.com/IgorShevchik.png" alt="Igor Shevchik" width="32" height="32" class="h-full w-full rounded-[inherit] object-cover"></span><span class="inline-flex items-center justify-center shrink-0 select-none overflow-hidden align-middle bg-base-100 dark:bg-base-900 size-5 text-4xs font-regular relative rounded-full ring-white dark:ring-base-700 last:me-0 ring -me-0.5"><img role="img" src="https://github.com/romhml.png" alt="Romain Hamel" width="32" height="32" class="h-full w-full rounded-[inherit] object-cover"></span><span class="inline-flex items-center justify-center shrink-0 select-none overflow-hidden align-middle bg-base-100 dark:bg-base-900 size-5 text-4xs font-regular relative rounded-full ring-white dark:ring-base-700 last:me-0 ring -me-0.5"><img role="img" src="https://github.com/noook.png" alt="Neil Richter" width="32" height="32" class="h-full w-full rounded-[inherit] object-cover"></span>
  <!--v-if-->
</div>"
`;

exports[`AvatarGroup > renders with size 3xl correctly 1`] = `
"<div class="inline-flex justify-end"><span class="inline-flex items-center justify-center shrink-0 select-none overflow-hidden align-middle bg-base-100 dark:bg-base-900 size-[94px] text-[34px] relative rounded-full ring-white dark:ring-base-700 last:me-0 ring-3 -me-2"><img role="img" src="https://github.com/IgorShevchik.png" alt="Igor Shevchik" width="32" height="32" class="h-full w-full rounded-[inherit] object-cover"></span><span class="inline-flex items-center justify-center shrink-0 select-none overflow-hidden align-middle bg-base-100 dark:bg-base-900 size-[94px] text-[34px] relative rounded-full ring-white dark:ring-base-700 last:me-0 ring-3 -me-2"><img role="img" src="https://github.com/romhml.png" alt="Romain Hamel" width="32" height="32" class="h-full w-full rounded-[inherit] object-cover"></span><span class="inline-flex items-center justify-center shrink-0 select-none overflow-hidden align-middle bg-base-100 dark:bg-base-900 size-[94px] text-[34px] relative rounded-full ring-white dark:ring-base-700 last:me-0 ring-3 -me-2"><img role="img" src="https://github.com/noook.png" alt="Neil Richter" width="32" height="32" class="h-full w-full rounded-[inherit] object-cover"></span>
  <!--v-if-->
</div>"
`;

exports[`AvatarGroup > renders with size 3xs correctly 1`] = `
"<div class="inline-flex justify-end"><span class="inline-flex items-center justify-center shrink-0 select-none overflow-hidden align-middle bg-base-100 dark:bg-base-900 size-xs2 text-4xs font-regular relative rounded-full ring-white dark:ring-base-700 last:me-0 ring -me-0.5"><img role="img" src="https://github.com/IgorShevchik.png" alt="Igor Shevchik" width="32" height="32" class="h-full w-full rounded-[inherit] object-cover"></span><span class="inline-flex items-center justify-center shrink-0 select-none overflow-hidden align-middle bg-base-100 dark:bg-base-900 size-xs2 text-4xs font-regular relative rounded-full ring-white dark:ring-base-700 last:me-0 ring -me-0.5"><img role="img" src="https://github.com/romhml.png" alt="Romain Hamel" width="32" height="32" class="h-full w-full rounded-[inherit] object-cover"></span><span class="inline-flex items-center justify-center shrink-0 select-none overflow-hidden align-middle bg-base-100 dark:bg-base-900 size-xs2 text-4xs font-regular relative rounded-full ring-white dark:ring-base-700 last:me-0 ring -me-0.5"><img role="img" src="https://github.com/noook.png" alt="Neil Richter" width="32" height="32" class="h-full w-full rounded-[inherit] object-cover"></span>
  <!--v-if-->
</div>"
`;

exports[`AvatarGroup > renders with size lg correctly 1`] = `
"<div class="inline-flex justify-end"><span class="inline-flex items-center justify-center shrink-0 select-none overflow-hidden align-middle bg-base-100 dark:bg-base-900 size-[42px] text-2xl relative rounded-full ring-white dark:ring-base-700 last:me-0 ring-2 -me-1.5"><img role="img" src="https://github.com/IgorShevchik.png" alt="Igor Shevchik" width="32" height="32" class="h-full w-full rounded-[inherit] object-cover"></span><span class="inline-flex items-center justify-center shrink-0 select-none overflow-hidden align-middle bg-base-100 dark:bg-base-900 size-[42px] text-2xl relative rounded-full ring-white dark:ring-base-700 last:me-0 ring-2 -me-1.5"><img role="img" src="https://github.com/romhml.png" alt="Romain Hamel" width="32" height="32" class="h-full w-full rounded-[inherit] object-cover"></span><span class="inline-flex items-center justify-center shrink-0 select-none overflow-hidden align-middle bg-base-100 dark:bg-base-900 size-[42px] text-2xl relative rounded-full ring-white dark:ring-base-700 last:me-0 ring-2 -me-1.5"><img role="img" src="https://github.com/noook.png" alt="Neil Richter" width="32" height="32" class="h-full w-full rounded-[inherit] object-cover"></span>
  <!--v-if-->
</div>"
`;

exports[`AvatarGroup > renders with size md correctly 1`] = `
"<div class="inline-flex justify-end"><span class="inline-flex items-center justify-center shrink-0 select-none overflow-hidden align-middle bg-base-100 dark:bg-base-900 size-8 text-sm relative rounded-full ring-white dark:ring-base-700 last:me-0 ring-2 -me-1.5"><img role="img" src="https://github.com/IgorShevchik.png" alt="Igor Shevchik" width="32" height="32" class="h-full w-full rounded-[inherit] object-cover"></span><span class="inline-flex items-center justify-center shrink-0 select-none overflow-hidden align-middle bg-base-100 dark:bg-base-900 size-8 text-sm relative rounded-full ring-white dark:ring-base-700 last:me-0 ring-2 -me-1.5"><img role="img" src="https://github.com/romhml.png" alt="Romain Hamel" width="32" height="32" class="h-full w-full rounded-[inherit] object-cover"></span><span class="inline-flex items-center justify-center shrink-0 select-none overflow-hidden align-middle bg-base-100 dark:bg-base-900 size-8 text-sm relative rounded-full ring-white dark:ring-base-700 last:me-0 ring-2 -me-1.5"><img role="img" src="https://github.com/noook.png" alt="Neil Richter" width="32" height="32" class="h-full w-full rounded-[inherit] object-cover"></span>
  <!--v-if-->
</div>"
`;

exports[`AvatarGroup > renders with size sm correctly 1`] = `
"<div class="inline-flex justify-end"><span class="inline-flex items-center justify-center shrink-0 select-none overflow-hidden align-middle bg-base-100 dark:bg-base-900 size-7 text-xs relative rounded-full ring-white dark:ring-base-700 last:me-0 ring -me-1.5"><img role="img" src="https://github.com/IgorShevchik.png" alt="Igor Shevchik" width="32" height="32" class="h-full w-full rounded-[inherit] object-cover"></span><span class="inline-flex items-center justify-center shrink-0 select-none overflow-hidden align-middle bg-base-100 dark:bg-base-900 size-7 text-xs relative rounded-full ring-white dark:ring-base-700 last:me-0 ring -me-1.5"><img role="img" src="https://github.com/romhml.png" alt="Romain Hamel" width="32" height="32" class="h-full w-full rounded-[inherit] object-cover"></span><span class="inline-flex items-center justify-center shrink-0 select-none overflow-hidden align-middle bg-base-100 dark:bg-base-900 size-7 text-xs relative rounded-full ring-white dark:ring-base-700 last:me-0 ring -me-1.5"><img role="img" src="https://github.com/noook.png" alt="Neil Richter" width="32" height="32" class="h-full w-full rounded-[inherit] object-cover"></span>
  <!--v-if-->
</div>"
`;

exports[`AvatarGroup > renders with size xl correctly 1`] = `
"<div class="inline-flex justify-end"><span class="inline-flex items-center justify-center shrink-0 select-none overflow-hidden align-middle bg-base-100 dark:bg-base-900 size-12 text-2xl relative rounded-full ring-white dark:ring-base-700 last:me-0 ring-3 -me-2"><img role="img" src="https://github.com/IgorShevchik.png" alt="Igor Shevchik" width="32" height="32" class="h-full w-full rounded-[inherit] object-cover"></span><span class="inline-flex items-center justify-center shrink-0 select-none overflow-hidden align-middle bg-base-100 dark:bg-base-900 size-12 text-2xl relative rounded-full ring-white dark:ring-base-700 last:me-0 ring-3 -me-2"><img role="img" src="https://github.com/romhml.png" alt="Romain Hamel" width="32" height="32" class="h-full w-full rounded-[inherit] object-cover"></span><span class="inline-flex items-center justify-center shrink-0 select-none overflow-hidden align-middle bg-base-100 dark:bg-base-900 size-12 text-2xl relative rounded-full ring-white dark:ring-base-700 last:me-0 ring-3 -me-2"><img role="img" src="https://github.com/noook.png" alt="Neil Richter" width="32" height="32" class="h-full w-full rounded-[inherit] object-cover"></span>
  <!--v-if-->
</div>"
`;

exports[`AvatarGroup > renders with size xs correctly 1`] = `
"<div class="inline-flex justify-end"><span class="inline-flex items-center justify-center shrink-0 select-none overflow-hidden align-middle bg-base-100 dark:bg-base-900 size-6 text-3xs font-regular relative rounded-full ring-white dark:ring-base-700 last:me-0 ring -me-0.5"><img role="img" src="https://github.com/IgorShevchik.png" alt="Igor Shevchik" width="32" height="32" class="h-full w-full rounded-[inherit] object-cover"></span><span class="inline-flex items-center justify-center shrink-0 select-none overflow-hidden align-middle bg-base-100 dark:bg-base-900 size-6 text-3xs font-regular relative rounded-full ring-white dark:ring-base-700 last:me-0 ring -me-0.5"><img role="img" src="https://github.com/romhml.png" alt="Romain Hamel" width="32" height="32" class="h-full w-full rounded-[inherit] object-cover"></span><span class="inline-flex items-center justify-center shrink-0 select-none overflow-hidden align-middle bg-base-100 dark:bg-base-900 size-6 text-3xs font-regular relative rounded-full ring-white dark:ring-base-700 last:me-0 ring -me-0.5"><img role="img" src="https://github.com/noook.png" alt="Neil Richter" width="32" height="32" class="h-full w-full rounded-[inherit] object-cover"></span>
  <!--v-if-->
</div>"
`;
